code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1227 @@
+	if(holotile && length(recent_speech)) //the proc should'nt be called if the list is empty in the first place but we might as well make sure.
+		holotile.say(recent_speech[pick(recent_speech)]) //say one of the 10 latest sentence said by the holo_host
+		addtimer(CALLBACK(src, .proc/holo_talk), 100, TIMER_OVERRIDE|TIMER_UNIQUE)
+
+
+///makes a hologram of the mob stepping on the tile, any new person stepping in will replace it with a new hologram
+/obj/effect/warped_rune/ceruleanspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(!istype(crossing,/mob/living) || cooldown > world.time)
+		return
+
+	if(locate(holotile) in rune_turf)//here to delete the previous hologram,
+		qdel(holotile)
+		holotile = null
+
+	cooldown = max_cooldown + world.time
+	holo_host = crossing
+	holotile = new(rune_turf) //setting up the hologram to look like the person that just stepped in
+	holotile.icon = holo_host.icon
+	holotile.icon_state = holo_host.icon_state
+	holotile.alpha = 100
+	holotile.name = "[holo_host.name] (Hologram)"
+	holotile.add_atom_colour("#77abff", FIXED_COLOUR_PRIORITY)
+	holotile.copy_overlays(holo_host, TRUE)
+
+
+//the code that follows is basically the code that changeling use to get people's last spoken sentences with a few tweaks.
+	recent_speech = list() //resets the list from its previous sentences
+	var/list/say_log = list()
+	var/log_source = holo_host.logging
+	for(var/log_type in log_source)
+		var/nlog_type = text2num(log_type)
+		if(nlog_type & LOG_SAY)
+			var/list/reversed = log_source[log_type]  //reverse the list so we get the last sentences instead of the first
+			if(islist(reversed))
+				say_log = reverseRange(reversed.Copy())
+				break
+
+	if(length(say_log) > 10) //we're going to get up to the last 10 sentences spoken by the holo_host
+		recent_speech = say_log.Copy(say_log.len - 11, 0)
+	else
+		for(var/spoken_memory in say_log)
+			if(recent_speech.len >= 10)
+				break
+			recent_speech[spoken_memory] = say_log[spoken_memory]
+
+	if(!length(recent_speech)) //lazy lists don't work here for whatever reason so we set it to null manually if the list is empty.
+		recent_speech = null
+		return
+
+	addtimer(CALLBACK(src, .proc/holo_talk), 100, TIMER_OVERRIDE|TIMER_UNIQUE)
+
+
+///destroys the hologram with the rune
+/obj/effect/warped_rune/ceruleanspace/Destroy()
+	qdel(holotile)
+	holotile = null
+	holo_host = null
+	recent_speech = null
+	return ..()
+
+
+/obj/item/slimecross/warping/pyrite
+	colour = "pyrite"
+	runepath = /obj/effect/warped_rune/pyritespace
+	effect_desc = "draws a rune that will randomly color whatever steps on it"
+
+
+/obj/effect/warped_rune/pyritespace
+	desc = "Who shall we be today? they asked, but not even the canvas would answer."
+	icon_state = "colorune"
+
+
+///colors whatever steps on the rune randomly
+/obj/effect/warped_rune/pyritespace/Crossed(atom/movable/colored)
+	. = ..()
+	colored.color = rgb(rand(0,255),rand(0,255),rand(0,255))
```

Comment:
This will color everything, including ghosts and whatnot.
Add some limits here.